src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Windows.cs

```diff
@@ -147,7 +148,10 @@ private static unsafe void Preallocate(string fullPath, long preallocationSize,
                     throw new IOException(SR.Format(errorCode == Interop.Errors.ERROR_DISK_FULL
                                                         ? SR.IO_DiskFull_Path_AllocationSize
                                                         : SR.IO_FileTooLarge_Path_AllocationSize,
-                                            fullPath, preallocationSize));
+                                            fullPath, preallocationSize), Win32Marshal.MakeHRFromErrorCode(errorCode));
+
+                    // SetFileInformationByHandle fails with ERROR_DISK_FULL in certain cases where the size is disallowed by filesystem
```

Comment:
nit: This comments is great and I would keep it, but IMO we should move it up. Right after the "// Only throw for errors that indicate there is not enough space." line?